app/services/api/requestsApi.ts

```diff
@@ -0,0 +1,41 @@
+import { api } from "./apiClient"
+import { ApiResponse } from "../../../backend/src/types/api.types"
+import { Leave, Assignment, Swap } from "../../../backend/src/types/requests.types"
+
+export const requestsApi = {
+  getLeaveRequests: async () => {
+    const response = await api.get<ApiResponse<Leave>>("/requests/leave")
```

Comment:
Gonna write the comment once but it should apply for all 3 leave/assignmentRequest/swapRequest APIS! super minor, just adjusting the shape to become a list so FE has a bit of an easier time

```suggestion
    const response = await api.get<ApiResponse<Leave[]>>("/requests/leave")
```